### PR TITLE
Fix the segcore payload reader can't read JSON data

### PR DESCRIPTION
**Pull request by yah01**
/kind bug
related #23389

### CODE REVIEW

**Comment by yah01** [comment]
related #24219

**Comment by yah01** [comment]
rerun ut

File: internal/core/src/storage/FieldDataInterface.h
Comment (congqixia) [inline internal/core/src/storage/FieldDataInterface.h:1]: why submit comment only file?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**